Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/README.md

```diff
@@ -0,0 +1,298 @@
+
+```
+dotnet tool install -g amazon.lambda.testtool --prerelease
+```
+
+To update the tool run the following command:
+
+```
+dotnet tool update -g amazon.lambda.testtool --prerelease
+```
+
+## Running the Test Tool
+
+### Lambda Emulator Mode
+Use this mode when you want to test Lambda functions directly without API Gateway integration.
+
+```
+# Start Lambda emulator on port 5050
+dotnet lambda-test-tool start --lambda-emulator-port 5050
+```
+
+### API Gateway Emulator Mode
+Use this mode when you want to test Lambda functions through API Gateway endpoints. This mode requires additional configuration through environment variables.
+
+```
+# Start API Gateway emulator on port 5051 in REST mode
+dotnet lambda-test-tool start \
+    --api-gateway-emulator-port 5051 \
+    --api-gateway-emulator-mode Rest
+
+```
+#### Required Configuration
+When running via command line, you must set the environment variable for API Gateway route configuration:
+
+Linux/macOS:
+```bash
+export APIGATEWAY_EMULATOR_ROUTE_CONFIG='{"LambdaResourceName":"AddLambdaFunction","HttpMethod":"Get","Path":"/add/{x}/{y}"}'
+```
+
+Windows (Command Prompt):
+
+```
+set APIGATEWAY_EMULATOR_ROUTE_CONFIG={"LambdaResourceName":"AddLambdaFunction","HttpMethod":"Get","Path":"/add/{x}/{y}"}
+```
+
+Windows (PowerShell):
+
+```
+$env:APIGATEWAY_EMULATOR_ROUTE_CONFIG='{"LambdaResourceName":"AddLambdaFunction","HttpMethod":"Get","Path":"/add/{x}/{y}"}'
+```
+
+### Combined Mode
+Use this mode when you want to run both Lambda and API Gateway emulators simultaneously.
+
+```
+# Start both emulators
+dotnet lambda-test-tool start \
+    --lambda-emulator-port 5050 \
+    --api-gateway-emulator-port 5051 \
+    --api-gateway-emulator-mode Rest
+```
+
+## Command Line Options
+| Option | Description | Required For |
+|--------|-------------|--------------|
+| `--lambda-emulator-port` | Port for Lambda emulator | Lambda Mode |
+| `--lambda-emulator-host` | Host for Lambda emulator | Lambda Mode |
+| `--api-gateway-emulator-port` | Port for API Gateway | API Gateway Mode |
+| `--api-gateway-emulator-mode` | API Gateway mode (Rest/HttpV1/HttpV2) | API Gateway Mode |
+| `--no-launch-window` | Disable auto-launching web interface | Optional |
+
+
+## API Gateway Configuration
+When using API Gateway mode, you need to configure the route mapping using the APIGATEWAY_EMULATOR_ROUTE_CONFIG environment variable. This can be a single route or an array of routes:
+
+### Single Route Configuration
+```
+{
+    "LambdaResourceName": "AddLambdaFunction",
+    "HttpMethod": "Get",
+    "Path": "/add/{x}/{y}"
+}
+```
+
+### Multiple Routes Configuration
+
+```
+[
+    {
+        "LambdaResourceName": "AddLambdaFunction",
+        "HttpMethod": "Get",
+        "Path": "/add/{x}/{y}"
+    },
+    {
+        "LambdaResourceName": "SubtractLambdaFunction",
+        "HttpMethod": "Get",
+        "Path": "/minus/{x}/{y}"
+    }
+]
+
+```
+
+## Running the Test Tool Project via an IDE
```

Comment:
Not sure I understand the point of this section. We don't expect users to download our repository and run it unless they want to contribute to the repo. In that case this would be information to provide in our contributing guide.

Reply:
I will move this to the contribution guide

Reply:
i moved this to a separate contribution guide